test/test_luaunit.lua

```diff
@@ -1947,6 +1948,8 @@ TestLuaUnitAssertionsError = {}
             id = 10,
         }) end)
 
+        lu.assertErrorMsgEquals("500", function() error(500) end)
+
```

Comment:
try with error(500, 2). I expect that this would fix the problem for lua 5.1 and 5.2 . At least, this is the only difference that I could spot versus self.f_with_error()